File: 04_analyze/GZlist_cum/code/analyze.R
library(tidyverse)


# data load 
GZcumplot <- read_csv(here::here("03_build/weekSIR/output/weekSIR.csv")) %>% 
  filter(pref == "Yamanashi") %>% 
  filter(week >= "2020-03-16")


# Cumulative number of GZ certified restaurants -------

gz_week_caseplot <- ggplot(data = GZcumplot,
                           mapping = aes(x = week,
                                         y = cumGZ))+
  xlab("Week") +
  ylab("Number of GZ-certified restaurants") +
  geom_point(size = 0.1, color = "forestgreen") +
  geom_line(size = 0.5, color = "forestgreen")  +
  scale_x_date(date_labels = "%Y-%m") +
  ggtitle("Time series of cumulative number of GZ-certified restaurants") +
  geom_vline(xintercept = as.Date("2020-07-17"), linetype = "dotdash", color = "forestgreen")+
  annotate("text",
           x = as.Date("2020-07-17"),
           y = Inf,
           label = "The GZ certification system",
           size = 4,
           vjust=5)+
  geom_label(aes(x = as.Date("2020-09-28"), y = 2500, label = "The first expansion"), fill = "white",
             size = 4, vjust=5) +
  geom_label(aes(x = as.Date("2021-04-01"),y = 3000, label = "The second expansion"), fill = "white",
             size = 4, vjust=5) +
  theme(title = element_text(size = 16),
        axis.text.x = element_text(size = 12),
        axis.title.x = element_text(size = 16), 
        axis.text.y = element_text(size = 12),
        axis.title.y = element_text(size = 16))

gz_week_caseplot

ggsave(here::here("04_analyze/GZlist_cum/output/cumGZ_timeseries.png"), gz_week_caseplot, width = 10, height = 5, dpi = 300)

File: 04_analyze/GZlist/code/analyze.py
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

# data load 
weekSIR = pd.read_csv("03_build/weekSIR/output/weekSIR.csv")

# data clean
weekSIR = weekSIR[weekSIR["pref"] == "Yamanashi"][["week", "cumGZ"]].iloc[20:] # since June 2020

series = pd.Series(weekSIR['cumGZ'], dtype='float') 
series.index = pd.to_datetime(weekSIR['week']) 

# plot 
series.plot(color = "green")
# plt.axvline("2020-07-17", color = 'green', linestyle = '--')
plt.annotate("The first expansion",
             xytext = ("2020-06-25", 1500),
             backgroundcolor = "lightgreen",
             xy = ("2020-09-25", 500),
             arrowprops = dict(arrowstyle='-|>'))
plt.annotate("The second expansion",
             xytext = ("2020-12-25", 1000),
             backgroundcolor = "lightgreen",
             xy = ("2021-02-25", 2500),
             arrowprops = dict(arrowstyle='-|>'))
plt.xlabel('Week')
plt.ylabel('Number of the GZ-certified restaurants')
plt.title("Time series of cumulative number of the GZ-certified restaurants")
plt.savefig("04_analyze/GZlist/output/cumulative_GZ_plot.png", format="png", dpi=250, pad_inches=0.1, bbox_inches="tight")
